src/player.cpp

```diff
@@ -11444,6 +11444,24 @@ action_id player::get_next_auto_move_direction()
     return get_movement_direction_from_delta( dp.x, dp.y, dp.z );
 }
 
+bool player::defer_move( tripoint next ) {
+    // next must be adjacent to current pos
+    tripoint dp = next - pos();
+    if( abs( dp.x ) > 1 || abs( dp.y ) > 1 || abs( dp.z ) > 1 ) {
```

Comment:
You can use `square_dist` for the adjacent test: `square_dist( ..., pos() ) == 1` means it's adjacent. That's slightly shorter, and you can reduce the second check to one `if` statement.

And your current checks interpret `next == pos()` as adjacent.

Reply:
Thanks for the tip, done.